crates/cxx-qt-gen/src/generator/cpp/threading.rs

```diff
@@ -19,18 +19,26 @@ pub fn generate(qobject_idents: &QObjectName) -> Result<GeneratedCppQObjectBlock
     result.forward_declares.push(format!(
         "using {cxx_qt_thread_ident} = ::rust::cxxqtlib1::CxxQtThread<{ident}>;"
     ));
+    // Ensure that the CxxQtThread<T> is of the correct size and alignment
+    // which should be two std::shared_ptr which are two size_t each
+    result.methods.push(CppFragment::Source(formatdoc! {
+        r#"
+        static_assert(alignof({cxx_qt_thread_ident}) <= sizeof(::std::size_t), "unexpected aligment");
+        static_assert(sizeof({cxx_qt_thread_ident}) == sizeof(::std::size_t[4]), "unexpected size");
```

Comment:
do we need more asserts for the constructor / move constructor stuff to check it's trivial ?

Reply:
I think we're doing the same that CXX does here, so it's good enough.